### PR TITLE
MINOR: add IQ docs to streams documentation

### PR DESCRIPTION
**Pull request by dguy**

### CODE REVIEW

**Comment by dguy** [comment]
@miguno @enothereska @guozhangwang @mjsax @bbejeck - this is mostly c&p from the confluent docs, with formatting etc.
Is there any specific way we link to code from the AK docs? i.e., so we don't need to change the version every time there is a release?

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk8-scala2.12/5878/
Test FAILed (JDK 8 and Scala 2.12).

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk7-scala2.11/5893/
Test PASSed (JDK 7 and Scala 2.11).

**Review by enothereska — APPROVED** [review]

**Review by mjsax — APPROVED** [review]
LGTM.
Don't think we have an easy way to dynamically link to code... I am not aware that plain HTML supports variables -- some JavaScript might do the trick, but not sure if we want to mess with it.

File: docs/streams/developer-guide.html [hunk 1]
```diff
@@ -15,9 +15,9 @@
  limitations under the License.
 -->
 
-<script><!--#include virtual="../js/templateData.js" --></script>
+<!--<script>&lt;!&ndash;#include virtual="../js/templateData.js" &ndash;&gt;</script>-->
```
Comment (guozhangwang) [inline docs/streams/developer-guide.html]: Is this intentional ?

File: docs/streams/developer-guide.html [hunk 2]
```diff
@@ -578,7 +1096,7 @@
         <a href="/{{version}}/documentation/streams/architecture" class="pagination__btn pagination__btn__prev">Previous</a>
         <a href="/{{version}}/documentation/streams/upgrade-guide" class="pagination__btn pagination__btn__next">Next</a>
     </div>
-</script>
+<!--</script>-->
```
Comment (guozhangwang) [inline docs/streams/developer-guide.html]: Ditto here.

**Review by guozhangwang — COMMENTED** [review]
Should we also update the `streams/index.html` page to include it int the table of contents?

File: docs/streams/developer-guide.html (hunk 1 above)
Comment (dguy) [reply docs/streams/developer-guide.html]: oops!

**Comment by dguy** [comment]
Thanks @guozhangwang updated

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk7-scala2.11/5904/
Test PASSed (JDK 7 and Scala 2.11).

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk8-scala2.12/5889/
Test PASSed (JDK 8 and Scala 2.12).

**Comment by guozhangwang** [comment]
@dguy @mjsax Regarding links to code, we have a `dotVersion` variable under `js/templateData.js` for that, usage example can be found here:

https://github.com/apache/kafka-site/blob/asf-site/0110/quickstart.html#L285

**Comment by guozhangwang** [comment]
But currently we do not have examples for IQ in apache kafka repo, so I will merge this PR as is and we can add the links when we have those under examples folder.

**Comment by guozhangwang** [comment]
Merged to trunk. @dguy Could you file another PR for `kafka-site` repo as well for 0110 folder so it can be reflected as well?

**Comment by guozhangwang** [comment]
@dguy cherry-picked to 0.11.0 as well after ec2faa179663b1d440e5de243063947c70df3837

**Comment by dguy** [comment]
@guozhangwang https://github.com/apache/kafka-site/pull/65

**Pull request closed without merging**